arch/x86/core/ia32/swap.S

```diff
@@ -87,6 +87,19 @@ SECTION_FUNC(TEXT, arch_swap)
 	pop %edx
 	pop %eax
 #endif
+
+
+#if defined(CONFIG_TRACING)
+
+	/* Save these as we are using to keep track of isr and isr_param */
+	pushl	%eax
+	pushl	%edx
+
+	call	sys_trace_thread_switched_out
+
+	popl	%edx
+	popl	%eax
```

Comment:
Can you please help me understand why we add `push/pop` of `%eax` here? I know it is another register (for x86), and you mention we want to keep `isr` and `isr_param`, which would need 2 registers. How did you know to use `%eax`?

Reply:
> These same complaints apply to the `CONFIG_EXECUTION_BENCHMARKING` block immediately before it that is already in master branch. Which also brings up the question on why we have two instrumentation points right after one another.

yep, this is going away, there will be only one instrumentation point and we will use tracing for measuring timing for the benchmarks we have right now.